### DISCUSSION TITLE
Bridge Relayer Running Modes and Indexing Model

### DISCUSSION THREAD

**Discussion by l-monninger** [movementlabsxyz/movement#884] [Ideas]
# Summary
In discussion of #794 , several alternative models for indexing events were proposed. Some couple nicely with different notions of running modes for the relayer itself. I'd like to open up this thread to discuss these running modes, particularly as the might concern applications such as the indexer and the Informer from [MIP-50](https://github.com/movementlabsxyz/MIP/pull/50).

To begin, I identify the following models:
- **Relayer Structured Logs**: in addition to performing the essential tasks of relaying, the relayer also writes structured logs of events and similar into a database. This is conceptually similar to the Aptos Indexer Processor.
   - A slight improvement on this model may be to emit a stream of events than can be written 
- **Light Relayer**: the relayer can opt into a mode where it simply listens for events, i.e., does not send transactions of its own. This allows us to reuse the application specific logic of the relayer to form other services without running into complications of multiple replicas or requiring every replica of a relayer to be funded for actually performing relaying. 
- **No Data Domain Duties**: the relayer is not written to perform any data domain responsibilities. These are instead achieved by composing other indexing services. In some ways, the **Light Relayer** concept can be seen as endorsing this notion; if features for the Data Domain of the **Light Relayer** mode do are not kep in the **Full Relayer**, then the **Full Relayer** has no data domain duties.

**Comment by l-monninger** [comment]
I would suggest a simpler way to view this is:
- **Relayer Mode:** the process has relaying but no indexing duties.
- **Indexer-Relayer Mode:** the process has both relaying and indexing duties, and indexes what the relayer has done.
- **Indexer Mode:** the process indexing but no relaying duties--simply indexing on-chain events.

**Reply by musitdev** [reply]
I agree with this model. It's allow to deploy the relayer in production without the indexer logic that can cause troubles on the Tx relaying, deploy an indexer that write all transfer events to be able to query bridge activity and run a test instance with both and relayer activity for testing purpose.
